### PR TITLE
Always setup warmup broker in e2e tests to preserve broker data plane logs

### PR DESCRIPTION
**Pull request by yolocs**
Fixes https://github.com/google/knative-gcp/issues/1417

## Proposed Changes

-
-
-

**Release Note**

```release-note
NONE
```

**Docs**

### CODE REVIEW

File: test/e2e-common.sh [hunk 1]
```diff
@@ -102,6 +104,22 @@ function enable_monitoring(){
...
+# propagated. A warm-up broker will force the data plane to be created before
+# the real testing. This helps prevent the credential propagation delay causing
+# test flakiness.
+#
+# 2. The broker data plane will be GCed if there is no broker. Usually this would
+# happen before we dump all the pod logs in the cloud-run-events namespace. The
+# warm-up broker makes sure there is always one broker left and thus data plane pods
+# won't be deleted before we dump logs.
+function warmup_broker_setup(){
+  ko apply -f ${CONFIG_WARMUP_GCP_BROKER}
```
Comment (Harwayne) [inline test/e2e-common.sh:120]: Should we clean this up at the end of the tests?

Comment (yolocs) [reply test/e2e-common.sh:120]: We don't reuse clusters. So I think it's fine to leak in-cluster resources.

Comment (yolocs) [reply test/e2e-common.sh:120]: Yes, we do. Let me do that.

Comment (yolocs) [reply test/e2e-common.sh:120]: @Harwayne Just confirmed with Chi. We can't reliably do resource cleanup with scripts in prow. What he did was to always delete all pubsub topics/subs before tests (https://github.com/google/knative-gcp/blob/master/test/e2e-secret-tests.sh#L85-L89).

**Comment by yolocs** [comment]
Will force push this PR as the only test failure is broker metrics and it was _not_ because target not receiving the event.

**Pull request merged**